Update optional-features.md to include no-implicit-route-model (#2007)
ember-learn/guides-source: guides/release/configuring-ember/optional-features.md

Available features:

>   no-implicit-route-model (Default: false)

Author: ```suggestion
  no-implicit-route-model (Default: true)
```

>     Removes the default record loading behaviour on Ember's Route.

Reviewer: ```suggestion
    Removes the default record loading behavior on Ember's Route.
```

This suggestion makes me sad 🙃 🇮🇪

for a real feature name when you run this command.

```bash
$ ember feature:disable no-implicit-route-model
Disabled no-implicit-route-model. Be sure to commit config/optional-features.json to source control!
```

As we can see from the warning, `@ember/optional-features` has created a file in `config/optional-features.json` to store the configuration for your project.
We commit it to our repository and we are off to the races!

### no-implicit-route-model

> This feature is related to esoteric features of route model loading that you likely do not use, or know exist, and [have been deprecated](https://deprecations.emberjs.com/v5.x#toc_deprecate-implicit-route-model) in `5.3.0`. They are due to be removed in `6.0.0`. To clear the deprecation, you can enable this feature.

Reviewer: ```suggestion
This feature is related to esoteric features of route model loading that you likely do not use, or know exist, and [have been deprecated](https://deprecations.emberjs.com/id/deprecate-implicit-route-model/) in `5.3.0`. They are due to be removed in `6.0.0`. To clear the deprecation, you can enable this feature.
```

With this feature disabled, Ember will automatically load a route's model if the `model` hook has not been implemented. In this case, Ember will attempt to try a few things before rendering this route's template.

1. If there is a `store` property on your route, it will attempt to call its `find` method. Assuming you have ember-data installed, you may be expecting this. The arguments will be extracted from the params. For example, if a dynamic segment is `:post_id`, there exists logic to split on the underscore and find a record of type post.

2. As a fallback, it will attempt to define a `find` method and use your Model instance's `find` method to fetch. If a Model cannot be found or if the found Model does not have a find method, an assertion is thrown.

Enabling this optional feature will remove this implicit model loading behavior and leave it to you to implement if and when you need it.

> <!-- eof - needed for pages that end in a code block  -->

Reviewer: ```suggestion

```
